<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script>
        // setTimeout(() => {
        //     console.log("hello");
        // }, 1000);

        //参数->函数(resolve,reject)
        //resolve和reject本身又是函数
        new Promise((resolve, reject) => {
            //第一次网络请求的代码
            setTimeout(() => {
                resolve()
            }, 1000)
        }).then(() => {
            //第一次拿到结果的处理代码
            console.log("hello");
            console.log("hello");
            console.log("hello");
            console.log("hello");
            return new Promise((resolve, reject) => {
                setTimeout(() => {
                    resolve()
                }, 1000)
            }).then(() => {
                console.log("vue");
                console.log("vue");
                console.log("vue");
                console.log("vue");
                return new Promise((resolve, reject) => {
                    setTimeout(() => {
                        resolve()
                    }, 1000)
                }).then(() => {
                    console.log("php");
                    console.log("php");
                    console.log("php");
                    console.log("php");
                })
            })
        })

        //什么情况下需要使用promise？一般情况下是有异步操作时，使用promise对这个异步操作进行封装
        new Promise((resolve, reject) => {
                //成功的时候调用resolve
                resolve("csy");
                //失败的时候调用reject
                reject("err")
            }).then((data) => {
                console.log(data)
            }).catch(err => {
                console.log(err);
            })
            //promise的另一种写法：then里面有两个函数参数，把错误的信息函数也写在then里面
        new Promise((resolve, reject) => {
            setTimeout(() => {
                resolve("aaa");
                reject("bbb");
            }, 2000)
        }).then(data => {
            console.log(data)
        }, err => {
            console.log(err);
        })
    </script>
</body>

</html>